test/utils/lazy.test.ts

```diff
@@ -0,0 +1,117 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import { Lazy, AsyncLazy } from "../../extension.bundle";
+import { delay } from '../../extension.bundle';
+
+suite('(unit) Lazy tests', () => {
+    suite('Lazy<T>', () => {
+        test('Normal', async () => {
+            let factoryCallCount = 0;
+            const lazy: Lazy<boolean> = new Lazy(() => {
+                factoryCallCount++;
+                return true;
+            });
+
+            lazy.value;
+            lazy.value;
+
+            assert.equal(factoryCallCount, 1, 'Incorrect number of value factory calls.');
+        });
+
+        test('With lifetime', async () => {
+            let factoryCallCount = 0;
+            const lazy: Lazy<boolean> = new Lazy(() => {
+                factoryCallCount++;
+                return true;
+            }, 5);
+
+            lazy.value;
+            lazy.value;

```

Comment:
I assume the second call (line 33) is completed within 5 ms. Otherwise the line 35 assert will fail. Should the lifetime be increased to larger number?

Reply:
Yes, it will be near-instant since it will go through the first `if` block: `if (this._isValueCreated) { ... }`